### PR TITLE
Edit post

### PR DESCRIPTION
**Pull request by umaslakshmi**
Still need to discuss strategies for saving past slugs

### CODE REVIEW

File: BlogTemplate/Pages/Edit.cshtml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+@addTagHelper "*, Microsoft.AspNetCore.Mvc.TagHelpers"
+
+@{
+    ViewData["Title"] = "Edit Post";
+}
+
+<article>
+    <h2>Edit Post</h2>
+    <form method="post">
+        <p><h3><label for="Title">Title</label></h3></p>
```
Comment (jimmylewis) [inline BlogTemplate/Pages/Edit.cshtml]: &lt;h3> doesn't really belong in a &lt;p>.  Neither do the &lt;input> fields below.

File: BlogTemplate/Pages/Edit.cshtml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        <p><input type="text" name="Title" asp-for="newPost.Title" /></p>
+        <p><h4><label for="Body">Body</label></h4></p>
+        <p><textarea name="Body" asp-for="newPost.Body"></textarea></p>
+        @*<p><input type="text" name="Body" asp-for="newPost.Body" value="@Model.oldPost.Body" /></p>*@
```
Comment (jimmylewis) [inline BlogTemplate/Pages/Edit.cshtml]: Don't check in commented-out code, unless it's serving as an example snippet.

File: BlogTemplate/Pages/Edit.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            if(oldPost == null)
+            {
+                RedirectToPage("/Index");
+            }
+        }
+
+        public IActionResult OnPostPublish()
+        {
+            newPost.IsPublic = true;
+            UpdatePost(newPost, oldPost);
```
Comment (jimmylewis) [inline BlogTemplate/Pages/Edit.cshtml.cs]: In both of the OnPost* handlers, is oldPost getting bound?  I didn't see any fields in the form for it to bind with...

File: BlogTemplate.Tests/Models/BlogDataStoreTests.cs [hunk 4]
```diff
@@ -234,6 +234,82 @@ public void GetAllPosts_ReturnsList()
...
+            {
+                Slug = "New-Title",
+                Title = "New Title",
+                Body = "New body",
+                IsPublic = true,
+                Excerpt = "New excerpt"
+            };
+
+            testDataStore.SavePost(oldPost);
+            testDataStore.SaveComment(comment, oldPost);
```
Comment (jimmylewis) [inline BlogTemplate.Tests/Models/BlogDataStoreTests.cs]: I talked to Lucas about removing this.  testDataStore.SavePost() will save all of the comments as well, so you could do oldPost.Comments.Add(comment) before calling SavePost.

Or, you could do it inline
```
Post oldPost = new Post
{
    Slug = "..."
    Comments = new List<Comment>
    {
        new Comment
        {
            AuthorEmail = "johnsmith@example.com",
            AuthorName = "John Smith",
            Body = "Sample Comment",
        }
    }
}
```

Comment (umaslakshmi) [inline BlogTemplate.Tests/Models/BlogDataStoreTests.cs]: will be addressed in Lucas's branch

**Pull request merged**